### PR TITLE
WIP: A first stab at analytic opacities

### PR DESCRIPTION
**Pull request by Yurlungur**
This is a first pass at analytic opacities. I'm not done, and I also haven't fully tested the machinery. But I wanted to get thoughts on what I have so far. In particular, I wanted your opinion on the function signatures, the structure of the variant, and the emphasis on header-only calls.

Note the function signatures don't expose Ye but they do expose radiation type. My reasoning for allowing this is that in principle one could run, e.g., neutrinos and photons at the same time, or some other kind of radiation. I'm not necessarily attached to this. The radiation type could be moved into the lambda and hidden, though it would be a little gross.

### CODE REVIEW

**Comment by Yurlungur** [comment]
Pinging @jdolence @dholladay00 @brryan

File: opacities/gray_opacity.hpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include <opac-utils/physical_constants.hpp>
+#include <ports-of-call/portability.hpp>
+
+#include "thermal_distributions.hpp"
+
+namespace singularity {
+
+template <typename ThermalDistribution, int NSPECIES> class GrayOpacity {
+public:
+  GrayOpacity(const Real kappa) : kappa_(kappa) {}
```
Comment (brryan) [inline opacities/gray_opacity.hpp:33]: kappa is often an opacity (dimensions cm^2 g^-1) whereas here kappa is an absorption coefficient (dimensions cm^-1). My fault for having done this in earlier code I think. If we're passing in opacities should not `EmissivityPerNuOmega` et al. then return `rho*kappa_*Bnu`?

You could add a separate class, `GrayAbsortionCoefficient`, for the most extreme approximators who want constant optical depth over some distance regardless of material density

Comment (brryan) [reply opacities/gray_opacity.hpp:33]: On second thought implementing `GrayAbsorptionCoefficient` would probably be a waste of time.

Comment (Yurlungur) [reply opacities/gray_opacity.hpp:33]: >  If we're passing in opacities should not EmissivityPerNuOmega et al. then return `rho*kappa_*Bnu`?

Good catch. This is a bug. Thanks. I don't think we need a `GrayAbsorptionCoefficient` though.

**Comment by Yurlungur** [comment]
Superseded by #3.

**Pull request closed without merging**